cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/RollingLocationLogAppender.java

```diff
@@ -173,21 +171,17 @@ public void close() throws IOException {
     }
   }
 
-  private void closeInvalidStreams() throws IOException {
-    if (!locationManager.getInvalidOutputStreams().isEmpty()) {
-      Iterator<Map.Entry<LocationIdentifier, LocationOutputStream>> iter = locationManager.getInvalidOutputStreams()
-        .entrySet().iterator();
-
-      while (iter.hasNext()) {
-        Map.Entry<LocationIdentifier, LocationOutputStream> entry = iter.next();
-        try {
-          entry.getValue().close();
-          iter.remove();
-        } catch (IOException e) {
-          LOG.warn("Exception while closing invalid output stream for {}, will retry to close it later",
-                   entry.getKey(), e);
-          throw e;
-        }
+  private void closeInvalidStream() throws IOException {
+    if (locationManager.getInvalidOutputStream() != null) {
+      LocationOutputStream invalidOutputStream = locationManager.getInvalidOutputStream();
+      try {
+        invalidOutputStream.close();
+        // because close was successful make this output stream null
+        invalidOutputStream = null;
```

Comment:
this is a local variable, you have to do `locationManager.setInvalidOutputStream(null)`